### PR TITLE
Add conversion Personalization Tags into HTML comments [MAILPOET-6376]

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

This PR converts written Personalization Tags registered in the email editor into HTML Comments, which should improve user experience.

## Code review notes

As part of those changes, it inserts Personalization tags with their attributes. The value for inserting is composed but can be overridden as a constructor parameter.

## QA notes

How to test it:
- Known personalization tags can be written, and they should be converted into HTML comments, which are highlighted with an inverted background and font color.
- The conversion happens in the subject, the preheader, and blocks.

Known issue:
- When a conversion happens, the cursor is moved to the start of the rich text. This is something that I was not able to solve, but I would like to check again later.
 
## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6376]

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-6376]: https://mailpoet.atlassian.net/browse/MAILPOET-6376?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:email-editor/written-tags)

_The latest successful build from `email-editor/written-tags` will be used. If none is available, the link won't work._

### CODE REVIEW

File: packages/js/email-editor/src/blocks/core/rich-text.tsx [hunk 1]
```diff
@@ -121,4 +126,82 @@ function extendRichTextFormats() {
...
+		const { content } = attributes;
+
+		// Fetch the personalization tags list
+		const list = useSelect(
+			( select ) => select( storeName ).getPersonalizationTagsList(),
+			[]
+		);
+
+		// Memoized function to replace content tags
+		const updatedContent = useCallback( () => {
```
Comment (costasovo) [inline packages/js/email-editor/src/blocks/core/rich-text.tsx]: I suggest changing the name of the function to `updateContent`. The current `updatedContent` fits better for a variable. What do you think?

File: packages/js/email-editor/src/blocks/index.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import { disableImageFilter, hideExpandOnClick } from './core/image';
 import {
 	disableCertainRichTextFormats,
 	extendRichTextFormats,
+	replaceWrittenPersonalizationTags,
```
Comment (costasovo) [inline packages/js/email-editor/src/blocks/index.ts]: What about changing the name to `activatePersonalizationTagsReplacing`? The current name sounds to me more like a one-time action.

**Review by costasovo — COMMENTED** [review]
@lysyjan Looks good! I left just a couple of comments related to naming. Really tiny details.

**Comment by lysyjan** [comment]
@costasovo I like your suggestions and updated the PR 👍

**Review by costasovo — APPROVED** [review]
@lysyjan Thank you for the updates!

**Pull request merged**